lib/platform/gitlab/index.js

```diff
@@ -323,29 +323,32 @@ async function ensureCommentRemoval() {
   // Todo: implement. See GitHub API for example
 }
 
+async function getPrList() {
+  if (!config.prList) {
+    const urlString = `projects/${config.repoName}/merge_requests?per_page=100`;
+    const res = await get(urlString, { paginate: true });
+    config.prList = res.body.map(pr => ({
+      number: pr.iid,
+      branchName: pr.source_branch,
+      title: pr.title,
+      state: pr.state === 'opened' ? 'open' : 'closed',
+      isClosed: pr.state !== 'opened',
+    }));
+  }
+  return config.prList;
+}
+
 async function findPr(branchName, prTitle, state = 'all') {
   logger.debug(`findPr(${branchName}, ${prTitle}, ${state})`);
-  const urlString = `projects/${config.repoName}/merge_requests?state=${
-    state
-  }&per_page=100`;
-  const res = await get(urlString, { paginated: true });
-  let pr = null;
-  res.body.forEach(result => {
-    if (
-      (!prTitle || result.title === prTitle) &&
-      result.source_branch === branchName
-    ) {
-      pr = result;
-      // GitHub uses number, GitLab uses iid
-      pr.number = pr.iid;
-      pr.body = pr.description;
-      pr.displayNumber = `Merge Request #${pr.iid}`;
-      if (pr.state !== 'opened') {
-        pr.isClosed = true;
-      }
-    }
-  });
-  return pr;
+  const prList = await getPrList();
+  return prList
+    .reverse()
```

Comment:
are you sure we should be using `reverse()`. I think usually we'd be fastest to search from newest to oldest, and GitLab's default sort is desc:

![image](https://user-images.githubusercontent.com/6311784/32765081-2e818d2a-c909-11e7-8dec-5dde532403c3.png)